datasets/machine-learning-dataset-for-sdo/sdo_ml_indexer.py

```diff
@@ -0,0 +1,85 @@
+from pathlib import Path
+import os
+import tarfile
+import csv
+import datetime as dt
+import argparse
+
+#
+#    small data parsing utility for the Machine Learning Dataset for SDO
+#
+
+
+def extract(file_names, target_dir):
+    for idx, file_name in enumerate(file_names):
+
+        print(f"processing {file_name}, {idx+1} of {len(file_names)}")
+        parts = file_name.name.split("_")
+        instrument = parts[0]
+        channel = parts[1]
+        year = parts[2][0:4]
+        month = parts[2][4:6]
+
+        with tarfile.open(file_name) as tar_file:
+            tar_file.extractall(target_dir / Path(year))
```

Comment:
does it make sense to have a folder structure like this

/<year>/<month>/<day>/<instrument><date>_<time>_<channel>.npz?